xdsl/parser.py

```diff
@@ -727,11 +730,6 @@ def parse_optional_attribute(self,
 
         # Attribute with default format
         if parse_with_default_format:
-            if not issubclass(attr_def, ParametrizedAttribute):
```

Comment:
Why was this removed?
AFAIK, we need this, otherwise users will get an ugly error.

Reply:
It seems that no code goes through it.
Do you have an example?

Reply:
I added a fail-in-purpose test and reverted this change. Overall we need to have tests for cases like that.
Otherwise one may assume that this is redundant piece of code.